presto-main/src/test/java/com/facebook/presto/sql/planner/sanity/TestVerifyNoOriginalExpression.java

```diff
@@ -0,0 +1,293 @@
+        ProjectNode projectNode = builder.project(new Assignments(map), filterNode);
+        testValidation(projectNode);
+    }
+
+    @Test
+    public void testAggregation()
+    {
+        ImmutableList<VariableReferenceExpression> groupingKeys = ImmutableList.of(variableReferenceExpression);
+        int groupingSetCount = 1;
+        ImmutableMap<VariableReferenceExpression, SortOrder> orderings = ImmutableMap.of(variableReferenceExpression, SortOrder.ASC_NULLS_FIRST);
+        OrderingScheme orderingScheme = new OrderingScheme(
+                groupingKeys,
+                orderings);
+        ImmutableMap<VariableReferenceExpression, AggregationNode.Aggregation> aggregations = ImmutableMap.of(
+                variableReferenceExpression,
+                new AggregationNode.Aggregation(
+                        callExpression,
+                        Optional.of(callExpression),
+                        Optional.of(orderingScheme),
+                        false,
+                        Optional.of(new VariableReferenceExpression("orderkey", BIGINT))));
+        ImmutableSet<Integer> globalGroupingSets = ImmutableSet.of(1);
+        AggregationNode.GroupingSetDescriptor groupingSets = new AggregationNode.GroupingSetDescriptor(groupingKeys, groupingSetCount, globalGroupingSets);
+        ImmutableList<VariableReferenceExpression> preGroupedVariables = ImmutableList.of();
+        Optional<VariableReferenceExpression> hashVariable = Optional.of(variableReferenceExpression);
+        Optional<VariableReferenceExpression> groupIdVariable = Optional.of(variableReferenceExpression);
+        AggregationNode aggregationNode = new AggregationNode(
+                new PlanNodeId("1"),
+                valuesNode,
+                aggregations,
+                groupingSets,
+                preGroupedVariables,
+                AggregationNode.Step.SINGLE,
+                hashVariable,
+                groupIdVariable);
+        testValidation(aggregationNode);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testValidateForApplyFailed()
+    {
+        ImmutableMap<VariableReferenceExpression, RowExpression> map = ImmutableMap.of(variableReferenceExpression, castToRowExpression(new SymbolReference("count")));
+        Assignments assignments = new Assignments(map);
+        ImmutableList<VariableReferenceExpression> variableReferenceExpressions = ImmutableList.of(variableReferenceExpression);
+        ApplyNode applyNode = builder.apply(
+                assignments,
+                variableReferenceExpressions,
+                valuesNode,
+                valuesNode);
+        testValidation(applyNode);
+    }
+
+    @Test
+    public void testTableFinish()
+    {
+        TableFinishNode tableFinishNode = new TableFinishNode(
+                new PlanNodeId("1"),
+                valuesNode,
+                new TableWriterNode.WriterTarget(){
```

Comment:
What I meant was `TestingWriterTarget` is an existing class (in sql.planner) you can use. No need to create your own lambda class.

Reply:
> TestingWriterTarget

Oh, found it, thanks!